homeassistant/components/zha/update.py

```diff
@@ -93,127 +85,81 @@ async def async_update_data(self) -> None:
+        # Stay in an indeterminate state until we actually send something
+        if self.entity_data.entity.progress == 0:
+            return True
 
-        try:
-            result = await self.zha_device.device.update_firmware(
-                image=self._latest_firmware,
-                progress_callback=self._update_progress,
-            )
-        except Exception as ex:
-            raise HomeAssistantError(f"Update was not successful: {ex}") from ex
-
-        # If we tried to install firmware that is no longer compatible with the device,
-        # bail out
-        if result == Status.NO_IMAGE_AVAILABLE:
-            self._attr_latest_version = self._attr_installed_version
-            self.async_write_ha_state()
+        # Rescale 0-100% to 2-100% to avoid 0 and 1 colliding with None, False, and True
+        return int(math.ceil(2 + 98 * self.entity_data.entity.progress / 100))
 
-        # If the update finished but was not successful, we should also throw an error
-        if result != Status.SUCCESS:
-            raise HomeAssistantError(f"Update was not successful: {result}")
+    @property
+    def latest_version(self) -> str | None:
+        """Latest version available for install."""
+        return self.entity_data.entity.latest_version
 
-        # Clear the state
-        self._latest_firmware = None
-        self._attr_in_progress = False
-        self.async_write_ha_state()
+    @property
+    def release_summary(self) -> str | None:
+        """Summary of the release notes or changelog.
 
-    async def async_added_to_hass(self) -> None:
-        """Call when entity is added."""
-        await super().async_added_to_hass()
+        This is not suitable for long changelogs, but merely suitable
+        for a short excerpt update description of max 255 characters.
+        """
+        return self.entity_data.entity.release_summary
 
-        # OTA events are sent by the device
-        self.zha_device.device.add_listener(self)
-        self.async_accept_signal(
-            self._ota_cluster_handler, SIGNAL_ATTR_UPDATED, self.attribute_updated
-        )
+    @property
+    def release_url(self) -> str | None:
+        """URL to the full release notes of the latest version available."""
+        return self.entity_data.entity.release_url
 
-    async def async_will_remove_from_hass(self) -> None:
-        """Call when entity will be removed."""
-        await super().async_will_remove_from_hass()
-        self._attr_in_progress = False
+    # We explicitly convert ZHA exceptions to HA exceptions here so there is no need to
+    # use the `@convert_zha_error_to_ha_error` decorator.
+    async def async_install(
+        self, version: str | None, backup: bool, **kwargs: Any
+    ) -> None:
+        """Install an update."""
+        try:
+            await self.entity_data.entity.async_install(version=version, backup=backup)
+        except ZHAException as exc:
+            raise HomeAssistantError(exc) from exc
```

Comment:
This needs an user-facing error message + translations.

Reply:
Translations are missing, yes.
For now, the same user-facing English-only error message is thrown from ZHA e.g. [here](https://github.com/zigpy/zha/blob/378e53641d02a450a99a67dccf79e97324196d77/zha/application/platforms/update.py#L276).
It's just "rewrapped" in a `HomeAssistantError` here.